src/pydvl/valuation/methods/classwise_shapley.py

```diff
@@ -167,37 +106,53 @@ def __init__(
             raise ValueError("Scorer must be a ClasswiseScorer.")
         self.scorer: ClasswiseSupervisedScorer = utility.scorer
         self.is_done = is_done
-        self.progress = progress
+        self.tqdm_args: dict[str, Any] = {
+            "desc": f"{self.__class__.__name__}: {str(is_done)}"
+        }
+        # HACK: parse additional args for the progress bar if any (we probably want
+        #  something better)
+        if isinstance(progress, bool):
+            self.tqdm_args.update({"disable": not progress})
+        else:
+            self.tqdm_args.update(progress if isinstance(progress, dict) else {})
+        self.normalize_values = normalize_values
 
     def fit(self, data: Dataset):
         self.result = ValuationResult.zeros(
             # TODO: automate str representation for all Valuations
-            algorithm=f"classwise-shapley",
+            algorithm=f"{self.__class__.__name__}-{self.utility.__class__.__name__}-{self.sampler.__class__.__name__}-{self.is_done}",
```

Comment:
Regarding the TODO, have a look at `ToStringMixin` in the freshly released sensai-utils package: https://github.com/opcode81/sensAI-utils/blob/master/src/sensai/util/string.py#L133

Reply:
There is already a separate issue for this thing specifically #614